packages/homepage/content/docs/7-environment.md

```diff
@@ -0,0 +1,63 @@
+---
+title: 'Environment'
+authors: ['Saeris']
+description: 'CodeSandbox uses different execution environments based on what type of Template you use.'
+---
+
+## What is an Environment?
+
+When using CodeSandbox, your code is evaluated and run by one of our built-in execution environments. These execution environments come in one of two types: Client and Container.
+
+Client environments run entirely inside of your browser and will continue to bundle your code even when you lose your connection to our servers. Containers on the other hand are executed server-side inside of a Docker container instance.
+
+While Client Sandboxes have faster previews and offline availability, they lack in configurability. Container Sandboxes take longer to start up, but effectively behave much like a local development environment, allowing you to customize your build tools, set up servers to listen to incoming requests, etc.
+
+Below is a list of each of our environments.
+
+## Client Environments
+
+Client Sandboxes each have their own bundler attached to them which are configured to support a specific framework and emulate their official CLI tools. They are not one-to-one implementations and thus cannot be configured within CodeSandbox, however they are designed to mirror the default behavior of the framework. If your project requires advanced configuration, try using a Container Sandbox instead.
+
+Currently it is not possible to convert a Client Sandbox to a Container Sandbox.
+
+### Angular Bundler
+
+> Based on Angular CLI
+
+### CXJS Bundler
+
+> Based on CXJS
+
+### Dojo Bundler
+
+> Based on @dojo/cli-create-app
+
+### Preact Bundler
+
+> Based on Preact CLI
+
+### React Bundler
+
+> Based on Create React App
+
+### Reason Bundler
+
+>
+
+### Static
```

Comment:
This is an interesting one, I think we can actually safely say that this is a separate environment from client and server environments. So you have

- client environment
- server environment
- static environment

What do you think?

Reply:
Static is still under the "Client" category because it doesn't require a connection to our servers. True it doesn't have any bundler, but I think it's important that we keep the distinction between Client and Container. Plus Static is grouped under Client Templates in the modal.

Reply:
For the static template our API server serves the files, so it needs to have a network connection with the API server to request the files (eg. if you request from `static.codesandbox.io/image.png` it would first go to our API server which returns the right file). However, there is also no container attached, we just serve as a file store.

Reply:
We can do this later too, not necessary to split it out now. Food for thought!